### DISCUSSION TITLE
How to authenticate in the connection?

### DISCUSSION THREAD

**Discussion by Gabsop** [ra1u/redis-dart#58] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/50323251/157303877-5a36a48c-5c6e-4b5f-b186-ded7a49a3655.png)
I'm trying to connect to the Redis but I have no idea how do I put my username and password to enter with TLS.

**Comment by ra1u** [comment, marked as answer]
First command you send should be auth.


```Dart
conn.connectSecure(<args>).than((Command command){
     command.send_object(["AUTH","username","password"])
     .then((var response) {
       print(response);
       command.send_object(["SET","token","value]).then(
          (var response) => print(response));
     });
});
```
Or equivalently using async

```Dart
Command command = await conn.connectSecure(<args>);
var rsp_auth = await command.send_object(["AUTH","username","password"]);
print(rsp_auth);
var rsp_token = await command.send_object((["SET","token","value]));
print(rsp_token);
```

I did not compile this, so let me know how this works for you.

**Reply by Gabsop** [reply]
It worked! Thanks! I suggest that you add this to the doc, its my first time using redis and I guess beginners like me don't know that 😅

**Reply by ra1u** [reply]
Good idea. I can merge your pull request if you send it.

**Reply by Gabsop** [reply]
I made the change to de readme and send a pull request, can you check it out?
